pika/connection.py

```diff
@@ -1846,50 +1930,37 @@ def _on_connection_start(self, method_frame):
 
         """
         self._set_connection_state(self.CONNECTION_START)
-        if self._is_protocol_header_frame(method_frame):
-            raise exceptions.UnexpectedFrameError
-        self._check_for_protocol_mismatch(method_frame)
-        self._set_server_information(method_frame)
-        self._add_connection_tune_callback()
-        self._send_connection_start_ok(*self._get_credentials(method_frame))
-
-    def _on_connect_timer(self):
-        """Callback for self._connection_attempt_timer: initiate connection
-        attempt in the context of the event loop
 
-        """
-        self._connection_attempt_timer = None
-
-        error = self._adapter_connect()
-        if not error:
-            self._on_connected()
-            return
+        try:
+            if self._is_protocol_header_frame(method_frame):
+                raise exceptions.UnexpectedFrameError(method_frame)
+            self._check_for_protocol_mismatch(method_frame)
+            self._set_server_information(method_frame)
+            self._add_connection_tune_callback()
+            self._send_connection_start_ok(*self._get_credentials(method_frame))
+        except Exception as error:  # pylint: disable=W0703
```

Comment:
This turns expected exceptions in asynchronously-executing code into graceful `on_open_error_callback()` calls.